src/include/detail/linalg/tdb_io.h

```diff
@@ -375,64 +372,60 @@ auto sizes_to_indices(const std::vector<T>& sizes) {
  * @return a Matrix of the data
  */
 template <class T>
-auto read_bin_local(const std::string& bin_file, size_t subset = 0) {
-  if (!std::filesystem::exists(bin_file)) {
-    throw std::runtime_error("file " + bin_file + " does not exist");
-  }
-  auto file_size = std::filesystem::file_size(bin_file);
+auto read_bin_local(
+    const tiledb::Context& ctx,
+    const std::string& bin_file,
+    size_t subset = 0) {
+  auto vfs = tiledb::VFS{ctx};
+
+  std::uint32_t dimension = 0u;
```

Comment:
(Minor) Do we need `std::` here? We don't do this in the rest of the codebase. Same below.